### ISSUE TITLE
macOS Sonoma unable to build postgres 16

### ISSUE THREAD

**Issue by sato11** [smashedtoatoms/asdf-postgres#77]
**Describe the bug**
On macOS Sonoma 14.2.1, the attempt to install Postgres v16 through `asdf install postgres latest` fails, where the latest version is 16.1 as the date of this writing. It says icu library is missing:

```
  ...
  checking whether to build with ICU support... yes
  checking for icu-uc icu-i18n... no
  configure: error: ICU library not found
  If you have ICU already installed, see config.log for details on the
  failure.  It is possible the compiler isn't looking in the proper directory.
  Use --without-icu to disable ICU support.
  FAILED: install exited with an error
```

**Additional context**
One of the possible fixes is setting `PKG_CONFIG_PATH` as [this post](https://viggy28.dev/article/postgres-v16-icu-installation-issue/) suggests:

```
export PKG_CONFIG_PATH=/usr/local/opt/icu4c/lib/pkgconfig/
```

If you get icu4c through homebrew you can confirm the suggestion via `brew info icu4c`:

```
For pkg-config to find icu4c you may need to set:
  export PKG_CONFIG_PATH="/usr/local/opt/icu4c/lib/pkgconfig"
```

Also, this error keeps the nightly build from passing since 2023-09-15, a day after [the release of Postgres v16](https://www.postgresql.org/about/news/postgresql-16-released-2715/), although I omit the CI logs as they evaporate in time.

**Comment by gerhard** [comment]
I had to do the same on macOS Monterey v12.7.1 - thank you @sato11 for creating this issue!

```console
PKG_CONFIG_PATH="$(brew --prefix)/opt/icu4c/lib/pkgconfig" asdf install postgres 16.1
```

**Comment by wrburgess** [comment]
I've tried both solutions that @sato11 and @gerhard provided and I'm not getting any different behavior when attempting to update to Postgres 16.1

* chip - Apple M1
* macOS - Sonoma 14.2.1 (23C71)
* asdf - v0.13.1

```
Building with options: --prefix=/Users/johnsmith/.asdf/installs/postgres/16.1     --with-uuid=e2fs     --with-openssl     --with-zlib     --with-libraries='/opt/local/lib:/sw/lib:/usr/local/opt/openssl/lib:/opt/homebrew/opt/openssl/lib:/opt/homebrew/opt/openssl@3.2/lib:/opt/homebrew/opt/openssl@3/lib:/opt/homebrew/opt/openssl@1.1/lib:/opt/homebrew/openssl/lib:/opt/homebrew/libressl/lib:/opt/homebrew/lib:/usr/local/lib:/usr/lib'     --with-includes='/opt/local/lib:/sw/lib:/usr/local/opt/openssl/include:/opt/homebrew/opt/openssl/include:/opt/homebrew/include:/opt/homebrew/opt/openssl@3.2/include:/opt/homebrew/opt/openssl@3/include:/opt/homebrew/opt/openssl@1.1/include:/opt/homebrew/openssl/include:/opt/homebrew/libressl/include:/opt/homebrew/include:/usr/local/include:/usr/local/lib:/usr/lib'
...
checking for pkg-config... no
configure: WARNING: *** Include directory '/opt/local/lib does not exist.
configure: WARNING: *** Include directory /sw/lib does not exist.
configure: WARNING: *** Include directory /usr/local/opt/openssl/include does not exist.
configure: WARNING: *** Include directory /opt/homebrew/openssl/include does not exist.
configure: WARNING: *** Include directory /opt/homebrew/libressl/include does not exist.
configure: WARNING: *** Include directory /usr/local/include does not exist.
configure: WARNING: *** Include directory /usr/local/lib does not exist.
configure: WARNING: *** Include directory /usr/lib' does not exist.
configure: WARNING: *** Library directory '/opt/local/lib does not exist.
configure: WARNING: *** Library directory /sw/lib does not exist.
configure: WARNING: *** Library directory /usr/local/opt/openssl/lib does not exist.
configure: WARNING: *** Library directory /opt/homebrew/openssl/lib does not exist.
configure: WARNING: *** Library directory /opt/homebrew/libressl/lib does not exist.
configure: WARNING: *** Library directory /usr/local/lib does not exist.
configure: WARNING: *** Library directory /usr/lib' does not exist.
checking allow thread-safe client libraries... yes
checking whether to build with ICU support... yes
checking for icu-uc icu-i18n... no
configure: error: ICU library not found
If you have ICU already installed, see config.log for details on the
failure.  It is possible the compiler isn't looking in the proper directory.
Use --without-icu to disable ICU support.
```

**Comment by sato11** [comment]
@wrburgess In your case the problem seems to be with the presence or availability of pkg-config itself rather than how pkg-config should find icu libraries.

Normally I believe it is accessible like this excerpt from CI logs shows;

```
checking for pkg-config... /usr/local/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
```

while yours says it's not:

```
checking for pkg-config... no
```

I assume what you need is perhaps `brew install pkg-config` if you haven't got it, before moving on with our solutions?

**Comment by wrburgess** [comment]
@sato11 that did the trick. I guess I was still using the dependencies that worked for 15.5 but `brew install pkg-config` is required for 16.1, at least on a Mac M1.

Thank you!

**Comment by AJReade** [comment]
Had the pkg issue when that was fixed continuously getting this. Any help would be massively appreciated. 

```usr/bin/gcc -I/opt/homebrew/Cellar/unixodbc/2.3.12/include -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Werror=unguarded-availability-new -Wendif-labels -Wmissing-format-attribute -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -Wno-unused-command-line-argument -Wno-compound-token-split-by-macro -O2 -pthread -D_REENTRANT -D_THREAD_SAFE  -I../include -I../../../../src/interfaces/ecpg/include -I../../../../src/interfaces/libpq -I../../../../src/include -I/opt/homebrew/Cellar/icu4c/74.2/include -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk   -I/opt/homebrew/opt/openssl/include -I/opt/homebrew/include -I/opt/homebrew/opt/openssl@3.3/include -I/opt/homebrew/opt/openssl@3.2/include -I/opt/homebrew/opt/openssl@3.1/include -I/opt/homebrew/opt/openssl@3/include -I/opt/homebrew/opt/openssl@1.1/include -I/opt/homebrew/include -DSO_MAJOR_VERSION=3  -c -o informix.o informix.c
informix.c:97:14: error: use of undeclared identifier 'CDECIMALTYPE'
        if (risnull(CDECIMALTYPE, (char *) arg1) || risnull(CDECIMALTYPE, (char *) arg2))
                    ^
informix.c:97:54: error: use of undeclared identifier 'CDECIMALTYPE'
        if (risnull(CDECIMALTYPE, (char *) arg1) || risnull(CDECIMALTYPE, (char *) arg2))
                                                            ^
informix.c:138:12: error: use of undeclared identifier 'CDECIMALTYPE'
                rsetnull(CDECIMALTYPE, (char *) result);
                         ^
informix.c:185:11: error: use of undeclared identifier 'CDECIMALTYPE'
        rsetnull(CDECIMALTYPE, (char *) np);
                 ^
informix.c:186:14: error: use of undeclared identifier 'CSTRINGTYPE'
        if (risnull(CSTRINGTYPE, cp))
                    ^
informix.c:232:11: error: use of undeclared identifier 'CDECIMALTYPE'
        rsetnull(CDECIMALTYPE, (char *) np);
                 ^
informix.c:233:14: error: use of undeclared identifier 'CDOUBLETYPE'
        if (risnull(CDOUBLETYPE, (char *) &dbl))
                    ^
informix.c:254:11: error: use of undeclared identifier 'CDECIMALTYPE'
        rsetnull(CDECIMALTYPE, (char *) np);
                 ^
informix.c:255:14: error: use of undeclared identifier 'CINTTYPE'
        if (risnull(CINTTYPE, (char *) &in))
                    ^
informix.c:276:11: error: use of undeclared identifier 'CDECIMALTYPE'
        rsetnull(CDECIMALTYPE, (char *) np);
                 ^
informix.c:277:14: error: use of undeclared identifier 'CLONGTYPE'
        if (risnull(CLONGTYPE, (char *) &lng))
                    ^
informix.c:367:11: error: use of undeclared identifier 'CSTRINGTYPE'
        rsetnull(CSTRINGTYPE, (char *) cp);
                 ^
informix.c:368:14: error: use of undeclared identifier 'CDECIMALTYPE'
        if (risnull(CDECIMALTYPE, (char *) np))
                    ^
13 errors generated.
make[4]: *** [informix.o] Error 1
make[3]: *** [all-compatlib-recurse] Error 2
make[2]: *** [all-ecpg-recurse] Error 2
make[1]: *** [all-interfaces-recurse] Error 2
make: *** [all-src-recurse] Error 2
```

**Comment by Nezteb** [comment]
I recently ran into this issue after following [the recommendation in the readme](https://github.com/smashedtoatoms/asdf-postgres#mac):
```
> brew install gcc readline zlib curl ossp-uuid icu4c pkg-config
> export PKG_CONFIG_PATH="/opt/homebrew/bin/pkg-config:$(brew --prefix icu4c)/lib/pkgconfig:$(brew --prefix curl)/lib/pkgconfig:$(brew --prefix zlib)/lib/pkgconfig"
> asdf install postgres 16.8
...
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Werror=unguarded-availability-new -Wendif-labels -Wmissing-format-attribute -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-unused-command-line-argument -Wno-compound-token-split-by-macro -Wno-format-truncation -Wno-cast-function-type-strict -O2 -I../../src/port -DFRONTEND -I../../src/include -I/opt/homebrew/Cellar/icu4c@77/77.1/include -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.5.sdk   -I/opt/homebrew/opt/openssl/include -I/opt/homebrew/include -I/opt/homebrew/opt/openssl@3.5/include -I/opt/homebrew/opt/openssl@3.4/include -I/opt/homebrew/opt/openssl@3/include -I/opt/homebrew/include -I/usr/local/lib  -c -o quotes.o quotes.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Werror=unguarded-availability-new -Wendif-labels -Wmissing-format-attribute -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-unused-command-line-argument -Wno-compound-token-split-by-macro -Wno-format-truncation -Wno-cast-function-type-strict -O2 -I../../src/port -DFRONTEND -I../../src/include -I/opt/homebrew/Cellar/icu4c@77/77.1/include -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.5.sdk   -I/opt/homebrew/opt/openssl/include -I/opt/homebrew/include -I/opt/homebrew/opt/openssl@3.5/include -I/opt/homebrew/opt/openssl@3.4/include -I/opt/homebrew/opt/openssl@3/include -I/opt/homebrew/include -I/usr/local/lib  -c -o snprintf.o snprintf.c
snprintf.c:414:27: error: 'strchrnul' is only available on macOS 15.4 or newer [-Werror,-Wunguarded-availability-new]
  414 |                         const char *next_pct = strchrnul(format + 1, '%');
      |                                                ^~~~~~~~~
...
```

Despite the error, I'm on macOS 15.5. 🤔
